src/components/ec-file-upload/ec-file-upload.vue

```diff
@@ -0,0 +1,135 @@
+<template>
+  <div :data-test="$attrs['data-test'] ? `${$attrs['data-test']} ec-file-upload` : 'ec-file-upload'">
+    <label
+      v-if="label || note"
+      class="ec-file-upload__label"
+    >
+      <span
+        v-if="label"
+        data-test="ec-file-upload__label-text"
+        class="ec-file-upload__label-text"
+      >{{ label }}</span>
+
+      <span
+        v-if="note"
+        data-test="ec-file-upload__label-note"
+        class="ec-file-upload__label-note"
+      >{{ note }}</span>
+    </label>
+
+    <ec-file-dropzone
+      data-test="ec-file-upload__dropzone"
+      :is-disabled="isDisabled"
+      @change="onChange"
+    >
+      <slot
+        name="title"
+        slot="title"
+      />
+
+      <slot
+        name="subtitle"
+        slot="subtitle"
+      />
+    </ec-file-dropzone>
+
+    <div
+      data-test="ec-file-upload__help-text"
+      class="ec-file-upload__help-text"
+      :class="{ 'ec-file-upload__help-text--is-disabled': isDisabled }"
+    >
+      <slot name="help-text" />
+    </div>
+
+    <ec-file-list
+      data-test="ec-file-upload__list"
+      :items="fileList"
+      :is-delete-disabled="isDisabled"
+      @delete="onDelete"
+    />
+  </div>
+</template>
+
+<script>
+import EcFileDropzone from '../ec-file-dropzone';
+import EcFileList from '../ec-file-list';
+
+export default {
+  name: 'EcFileUpload',
+  components: {
+    EcFileDropzone,
+    EcFileList,
+  },
+  model: {
+    prop: 'value',
+    event: 'change',
+  },
+  props: {
+    value: {
+      type: Array,
+      default: () => ([]),
+    },
+    label: {
+      type: String,
+    },
+    note: {
+      type: String,
+    },
+    isDisabled: {
+      type: Boolean,
+    },
+  },
+  data() {
+    return {
+      fileList: [],
+    };
+  },
+  watch: {
+    value: {
+      immediate: true,
+      handler(newValue) {
+        this.fileList = newValue;
+      },
+    },
+  },
+  methods: {
+    onChange(files) {
+      this.fileList = [...files];
```

Comment:
Shouldn't this component be responsible for maintaining all the files in the list? Right now, instead of adding the files received with the event to the list, we are replacing the existing list. I thought that this would be the component that would add files to the list it manages and that the files would only be deleted if the user manually deletes them via the button

Reply:
We have 3 components
file-Dropzone: Allow the user to upload the files.
file list: Renders which files we have and if a user clicks to remove a file, it emits the item clicked to be delete.
file-upload: Is the core component that wraps the other two. The state is hold there. We upload there the data from dropzone and file list gets the data to render the list. File list acts as the medium of communication for the other two components. It also exposes a v-model with the data-hold.

Reply:
Yes, I agree with all that. Sorry if I didn't express myself well, but my comment was more in the sense that the `EcFileUpload` component is not handling the files received. It doesn't add them to the list of existing ones, but replaces the list. When an event with files is received from the `EcFileDropzone` component, the `EcFileUpload` component simply discards the existing files in the data `fileList` and replaces them with the new ones. I thought that when the component received an event with files from `EcFileDropzone` it would add it to the existing `fileList` data.

Reply:
Great, thank you for pointing it out @jafcalvente , I added the functionality. Pending some test for it which for now give me a bit of hard time :)